### PR TITLE
Feat - Change the update and delete logic

### PR DESCRIPTION
**Pull request by g-pelletier**

### CODE REVIEW

File: importer/lambda/cross_region_importer.py [hunk 1]
```diff
@@ -113,41 +118,42 @@ def _create_new_cross_stack_references(requested_exports, importer_context, tabl
...
         print(f'Removing cross-stack ref: {cross_stack_ref_id}')
-        cross_stack_ref_table.delete_item(
-            Key={'CrossStackRefId': cross_stack_ref_id},
-        )
+        try:
+            cross_stack_ref_table.delete_item(
+                Key={'CrossStackRefId': cross_stack_ref_id},
+                ConditionExpression=Attr('CrossStackRefId').eq(cross_stack_ref_id),
+            )
+        except ClientError:
```
Comment (spg) [inline importer/lambda/cross_region_importer.py]: ClientError can raised for many reasons (bad request, throttling, etc). You should make sure the `ClientError` you're catching has the proper error code you're expecting. If not, you should re-raise the exception.

File: importer/lambda/cross_region_importer.py [hunk 2]
```diff
@@ -113,41 +118,42 @@ def _create_new_cross_stack_references(requested_exports, importer_context, tabl
...
+                Key('ImporterStackId').eq(importer_context.stack_id) &
+                Key('ImporterLabel').eq(label) &
+                Key('ImporterLogicalResourceId').eq(importer_context.logical_resource_id)
+            )
+            cross_stack_ref_ids = [
+                cross_ref_id['CrossStackRefId'] for cross_ref_id in scan_response['Items']
+                if cross_ref_id != cross_stack_ref_id
+            ]
+
+            while scan_response.get('LastEvaluatedKey'):
```
Comment (spg) [inline importer/lambda/cross_region_importer.py]: Why is there a `while` loop here?

File: importer/lambda/cross_region_importer.py [hunk 3]
```diff
@@ -113,41 +118,42 @@ def _create_new_cross_stack_references(requested_exports, importer_context, tabl
...
+                scan_response = cross_stack_ref_table.scan(ExclusiveStartKey=scan_response['LastEvaluatedKey'])
+                cross_stack_ref_ids.extend([cross_ref_id['CrossStackRefId'] for cross_ref_id in scan_response['Items']])
+
+            for ref_id in cross_stack_ref_ids:
+                print(f'Deleting {ref_id}')
+                cross_stack_ref_table.delete_item(
```
Comment (spg) [inline importer/lambda/cross_region_importer.py]: What if throttling happens here?

**Comment by g-pelletier** [comment]
Done

**Pull request merged**